modules/oadp-api-details.adoc

```diff
@@ -0,0 +1,164 @@
+// Module included in the following assemblies:
+//
+// * oadp-api
+
+:_content-type: REFERENCE
+[id="oadp-api-details_{context}"]
+= OADP API
+
+The following tables expose the structure of the OADP API:
+
+.DataProtectionApplicationSpec
+[options="header"]
+|===
+|Property|Type|Description
+
+|`BackupLocations`
+|[] `BackupLocation`
+|Defines the list of desired configurations to use for `BackupStorageLocations`.
+
+|`SnapshotLocations`
+|[] `SnapshotLocation`
+|Defines the list of desired configurations to use for `VolumeSnapshotLocations`.
+
+|`UnsupportedOverrides`
+|map [ `UnsupportedImageKey` ] string
+|Can be used to override the deployed dependent images for development.
+
+|`PodAnnotations`
+|map [ string ] string
+|Used to add annotations to pods deployed by Operators.
+
+|`PodDnsPolicy`
+|[`corev1.DNSPolicy`] link:https://pkg.go.dev/k8s.io/api/core/v1#DNSPolicy[DNSPolicy]
+|Defines how the configuration of the DNS of a pod.
+
+|`PodDnsConfig`
+|[`corev1.PodDNSConfig`] link:https://pkg.go.dev/k8s.io/api/core/v1#PodDNSConfig[PodDNSConfig]
+|Defines the DNS parameters of a pod in addition to those generated from `DNSPolicy`.
+
+|`BackupImages`
+|*bool
+|Used to specify whether or not you want to deploy a registry for enabling backup and restore of images.
+
+|`Configuration`
+|*`ApplicationConfig`
+|Used to define the data protection application's server configuration.
+|===
+
+.BackupLocation
+[options="header"]
+|===
+|Property|Type|Description
+
+|`name`
+|`metav1. ObjectMeta`
+|Location name.
+
+|`velero`
+|*link:(https://velero.io/docs/v1.9/api-types/backupstoragelocation/[velero.BackupStorageLocationSpec]
+|Location to store volume snapshots. For further details, see link:https://github.com/openshift/oadp-operator/blob/076f1495f6de65b0e96392cb6ef7cff66cf7defb/docs/config/bsl_and_vsl.md[Backup Storage Locations and Volume Snapshot Locations Customization].
+|===
+
+.VolumeSnapshot
+[options="header"]
+|===
+|Property|Type|Description
+
+|`name`
+|`metav1. ObjectMeta`
+|Location name.
+
+|`velero`
+|*link:(https://velero.io/docs/v1.9/api-types/volumesnapshotlocation/[velero.VolumeSnapshotLocationSpec]
```

Comment:
maybe point to https://pkg.go.dev/github.com/vmware-tanzu/velero/pkg/apis/velero/v1#VolumeSnapshotLocationSpec instead